res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -0,0 +1,2227 @@
+            var b = parseInt(byteStr, 16);
+            if (b < 0 || b > 255) {
+                ok = false;
+                HIDDebug("number out of range: " + byteStr + " " + b);
+            }
+            data[i][j] = b;
+        }
+        if (ok) {
+            var header = 0x80 + i;
+            if (i === 2) {
+                header = 0xF4;
+            }
+            controller.send(data[i], data[i].length, header);
+        }
+    }
+    TraktorS3.kontrol.setInputLineMode(false);
+};
+
+TraktorS3.shutdown = function() {
+    // Deactivate all LEDs
+    var dataStrings = [
+        "      00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00 ",
+        "      00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00 " +
+        "00 00 00 00  00 00 00 00  00 00 00 00  00 00 00",
+    ];
+
+    var data = [Array(), Array()];
+
+
+    for (var i = 0; i < data.length; i++) {
+        var tokens = dataStrings[i].split(/\s+/);
+        data[i].length = tokens.length;
+        for (var j = 0; j < tokens.length; j++) {
+            var byteStr = tokens[j];
+            if (byteStr.length === 0) {
+                continue;
+            }
+            data[i][j] = parseInt(byteStr, 16);
+        }
+        controller.send(data[i], data[i].length, 0x80 + i);
+    }
+
+    HIDDebug("TraktorS3: Shutdown done!");
+};
+
+TraktorS3.init = function(_id) {
+    this.kontrol = new TraktorS3.Controller();
+    this.kontrol.Decks = {
+        "deck1": new TraktorS3.Deck(this.kontrol, 1, "deck1"),
+        "deck2": new TraktorS3.Deck(this.kontrol, 2, "deck2"),
+    };
+
+    this.kontrol.Channels = {
+        "[Channel1]": new TraktorS3.Channel(this.kontrol, this.kontrol.Decks["deck1"], "[Channel1]"),
+        "[Channel3]": new TraktorS3.Channel(this.kontrol, this.kontrol.Decks["deck1"], "[Channel3]"),
+        "[Channel4]": new TraktorS3.Channel(this.kontrol, this.kontrol.Decks["deck2"], "[Channel4]"),
+        "[Channel2]": new TraktorS3.Channel(this.kontrol, this.kontrol.Decks["deck2"], "[Channel2]"),
+    };
+
+    this.kontrol.fxController = new TraktorS3.FXControl(this.kontrol);
+
+    this.kontrol.registerInputPackets();
+    this.kontrol.registerOutputPackets();
+    HIDDebug("TraktorS3: Init done!");
+
+    if (TraktorS3.DebugMode) {
+        TraktorS3.debugLights();
+    } else {
+        this.kontrol.lightDeck("[Channel3]", false);
+        this.kontrol.lightDeck("[Channel4]", false);
+        this.kontrol.lightDeck("[Channel1]", false);
+        this.kontrol.lightDeck("[Channel2]", true);
```

Comment:
Is there any particular reason for this unusual order?

Reply:
yes!  comment added